apps/src/music/views/PackDialog2.tsx

```diff
@@ -0,0 +1,349 @@
+      aria-label={folder.name}
+      tabIndex={0}
+      role="button"
+      ref={isSelected ? currentFolderRefCallback : null}
+    >
+      <div>
+        {imageSrc && (
+          <div
+            className={classNames(
+              styles.packImageContainer,
+              isSelected && styles.packImageContainerSelected
+            )}
+          >
+            <img
+              className={styles.packImage}
+              src={imageSrc}
+              alt=""
+              draggable={false}
+            />
+            {false && imageAttributionAuthor && (
+              <div
+                className={classNames(
+                  styles.packImageAttribution,
+                  packImageAttributionLeft && styles.packImageAttributionLeft
+                )}
+                style={{color: imageAttributionColor}}
+              >
+                <FontAwesomeV6Icon
+                  iconName={'brands fa-creative-commons'}
+                  iconStyle="solid"
+                  className={styles.icon}
+                />
+                &nbsp;
+                <FontAwesomeV6Icon
+                  iconName={'brands fa-creative-commons-by'}
+                  iconStyle="solid"
+                  className={styles.icon}
+                />
+                &nbsp;
+                {imageAttributionAuthor}
+              </div>
+            )}
+          </div>
+        )}
+      </div>
+      <div
+        className={classNames(
+          styles.packName,
+          mode !== 'artist' && styles.packBold
+        )}
+      >
+        {folder.name}
+      </div>
+      {folder.artist && (
+        <div
+          className={classNames(
+            styles.packArtist,
+            mode === 'artist' && styles.packBold
+          )}
+        >
+          {folder.artist}
+        </div>
+      )}
+    </div>
+  );
+};
+
+interface PackDialogProps {
+  player: MusicPlayer;
+}
+
+type Mode = 'popular' | 'song' | 'artist';
+
+/**
+ * The PackDialog allows the user to preview and choose from the set of restricted
+ * sound packs.
+ */
+const PackDialog2: React.FunctionComponent<PackDialogProps> = ({player}) => {
+  const dispatch = useAppDispatch();
+
+  const currentPackId = useAppSelector(state => state.music.packId);
+
+  const library = MusicLibrary.getInstance();
+
+  // Use a ref for instant access to this value inside onPreview.
+  const playingPreview = useRef<string | null>(null);
+
+  const [mode, setMode] = useState<Mode>('popular');
+
+  // Use state so that we can re-render when the preview state changes.
+  const [playingPreviewState, setPlayingPreviewState] = useState<string | null>(
+    null
+  );
+
+  const [selectedFolderId, setSelectedFolderId] = useState<string | null>(null);
+
+  const currentFolderRef: React.MutableRefObject<HTMLDivElement | null> =
+    useRef(null);
+
+  const analyticsReporter = useContext(AnalyticsContext);
+
+  const onModeChange = useCallback((value: Mode) => {
+    setMode(value);
+  }, []);
```

Comment:
nit: just call `setMode` inline on the change handler instead of creating a separate callback?

Reply:
Good call.